### PR TITLE
chore: update all references to kn-plugin-func

### PR DESCRIPTION
**Pull request by lance**
# Changes

- 🧹 update all references to "kn-plugin-func", changing them to "func"
 
I did not change references in CHANGELOG.md but otherwise, I believe this should cover all configuration settings, image names, package names, etc.

Signed-off-by: Lance Ball <lball@redhat.com>

/kind chore

```release-note
changes package name from knative.dev/kn-plugin-func to knative.dev/func
```

### CODE REVIEW

**Comment by lance** [comment]
> Base: **47.20**% // Head: **49.10**% // Increases project coverage by **`+1.89%`** 🎉

Hilarious

**Comment by lance** [comment]
@dsimansk this verify failure I believe is from `hack/update-codegen.sh` on a mac :/

File: docs/reference/func_create.md [hunk 1]
```diff
@@ -31,16 +31,26 @@ DESCRIPTION
...
+	  go           knative/improve
+	  go           knative/uppercase
 	  node         cloudevents
 	  node         http
+	  nodejs       functastic/metacontroller
+	  nodejs       knative/metacontroller
 	  python       cloudevents
 	  python       http
+	  python       functastic/numpy
+	  python       knative/numpy
```
Comment (lkingland) [inline docs/reference/func_create.md:45]: Pretty sure these are not intended.  See #1320 and #1321

**Review by lkingland — COMMENTED** [review]
Looking great!
As you know the Knative dependency management tooling uses `golicenses` which seems to give slightly different results on different OSs.   This is the root cause of the `verify` CI task failure.  When checked in the CI environment, the changes to the `third_party` directory in this PR are not present when it does its self-check.
Therefore, until we can isolate the underlying cause and either fix our usage of golicenses to be non-os-depenent, the quick way to fix the error is to remove the folloiwing changes to the `third_party` directory:
- removal of `third_party/VENDOR-LICENSE/github.com/moby/sys/mount/LICENSE`
- rename of `third_party/VENDOR-LICENSE/github.com/opencontainers/selinux/go-selinux)/LICENSE`
- rename of `third_party/VENDOR-LICENSE/knative.dev/func/LICENSE`
If that doesn't do the trick, I have a small GitHub action I use temporarily to get some better log statements from that action to show the exact conflicts reported.

**Comment by lkingland** [comment]
> cause and either fix our usage of golicenses to be non-os-depenent,

Perhaps we should patch the Knative script which invokes `go-licenses` to include `GOOS=linux` such that others using non-linux OSs do not encounter the same problem.

File: cmd/root.go [hunk 2]
```diff
@@ -170,7 +170,7 @@ func deriveName(explicitName string, path string) string {
 
 // deriveNameAndAbsolutePathFromPath returns resolved function name and absolute path
 // to the function project root. The input parameter path could be one of:
-// 'relative/path/to/foo', '/absolute/path/to/foo', 'foo' or ''
+// 'relative/path/to/foo', '/absolute/path/to/foo', 'foo' or ”
```
[the remark this answers is not in the document]
Comment (lance) [reply cmd/root.go:173]: My IDE is a nightmare and keeps changing this. Ridonkulous!

**Comment by lance** [comment]
@lkingland @matejvasek @zroubalik ptal again 🙇

File: README.md [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
-[![License](https://img.shields.io/github/license/knative-sandbox/kn-plugin-func)](https://github.com/knative-sandbox/kn-plugin-func/blob/main/LICENSE)
-[![Releases](https://img.shields.io/github/v/release/knative-sandbox/kn-plugin-func.svg?label=Release)](https://github.com/knative-sandbox/kn-plugin-func/releases)
+[![CI Status](https://github.com/knative/func/actions/workflows/ci.yaml/badge.svg)](https://github.com/knative/func/actions/workflows/ci.yaml)
+[![Client API Documentation](https://pkg.go.dev/badge/knative.dev/func?utm_source=godoc)](https://pkg.go.dev/knative.dev/func)
+[![Issues](https://img.shields.io/github/issues/knative-sandbox/kn-plugin-func.svg)](https://github.com/knative/func/issues)
+[![License](https://img.shields.io/github/license/knative-sandbox/kn-plugin-func)](https://github.com/knative/func/blob/main/LICENSE)
+[![Releases](https://img.shields.io/github/v/release/knative-sandbox/kn-plugin-func.svg?label=Release)](https://github.com/knative/func/releases)
 [![codecov](https://codecov.io/gh/knative-sandbox/kn-plugin-func/branch/main/graph/badge.svg)](https://codecov.io/gh/knative-sandbox/kn-plugin-func)
```
Comment (zroubalik) [inline README.md:8]: there are some leftovers pointing to knative-sandbox/kn-plugin-func

**Comment by salaboy** [comment]
It will be great to merge this asap right?

**Pull request merged**